pkg/reflector/download.go

```diff
@@ -55,37 +70,49 @@ func (d *S3Downloader) DownloadTo(w io.Writer) (n int64, err error) {
 		// retry
 		return -1, errors.WithTypes(errors.Wrap(err, "get s3 data"), errs.ErrTypeTemporary)
 	}
-	defer obj.Body.Close()
-	compressedSize := obj.ContentLength
-	var reader io.Reader = obj.Body
+
 	if strings.HasSuffix(d.Key, ".gz") {
-		reader, err = gzip.NewReader(reader)
+		n, err = Unzip(file, w)
 		if err != nil {
-			return n, errors.Wrap(err, "create gzip reader")
+			return n, errors.Wrap(err, "unzip snapshot")
 		}
 	}
-	n, err = io.Copy(w, reader)
+
+	events.Log("LDB inflated %d -> %d bytes", numBytes, n)
+
+	return
+}
+
+func Unzip(src io.Reader, dest io.Writer) (int64, error) {
```

Comment:
Do you want to add a metric around this start and end time as well in case? I'm not sure if it would only be in the io.Copy part or not.